phase/configure_k0s.go

```diff
@@ -336,7 +336,7 @@ func (p *ConfigureK0s) configFor(h *cluster.Host) (string, error) {
 		}
 	}
 
-	if cfg.Dig("spec", "storage", "etcd", "peerAddress") != nil || h.PrivateAddress != "" {
+	if cfg.Dig("spec", "storage", "etcd", "peerAddress") != nil && h.PrivateAddress != "" {
```

Comment:
Hmm. I think this needs some more logic. Now it would never set the peerAddress if one wasn't set already.

- It should check if `spec.storage.type` is `etcd`.
- If role is single and storage type is not set, default is kine, otherwise the default is etcd.
- Overwriting a user defined peerAddress may be wrong in any case
- I'm not sure why this checks for `PrivateAddress` but uses `addr` 🤔